digest/digester.go

```diff
@@ -54,6 +54,15 @@ func (a Algorithm) String() string {
 	return string(a)
 }
 
+// Size returns number of bytes returned by the hash.
+func (a Algorithm) Size() int {
+	h, ok := algorithms[a]
+	if !ok {
+		return 0
```

Comment:
Tarsum has its own parse method so isn't affected(not better or worse). I can try to fix tarsum as well if needed, of course.